NFT/src/main.sw

```diff
@@ -0,0 +1,368 @@
+contract;
+
+dep contract_abi;
+dep data_structures;
+dep errors;
+dep events;
+
+use contract_abi::NFT;
+use data_structures::MetaData;
+use errors::{AccessError, ApprovalError, InitError, InputError};
+use events::{ApprovalEvent, BurnEvent, MintEvent, OperatorEvent, TransferEvent};
+
+use std::{
+    assert::require,
+    chain::auth::{AuthError, msg_sender},
+    hash::sha256,
+    identity::Identity,
+    logging::log,
+    option::Option,
+    result::Result,
+    storage::StorageMap,
+    vec::Vec,
+};
+
+storage {
+    /// Determines if only the `admin` is allowed to call the mint function.
+    /// This is only set on the initalization of the contract.
+    access_control: bool,
+
+    /// Stores an `Option` of the `Identity` that has permission to mint if `access_control` is set to true.
+    /// Will store `None` if this contract does not have `access_control` set.
+    /// Only the `admin` is allowed to change the `admin` of the contract.
+    admin: Option<Identity>,
+
+    /// Used of O(1) lookup of the number of tokens owned by each `Identity`.
```

Comment:
```suggestion
    /// Used for O(1) lookup of the number of NFT tokens owned by each `Identity`.
```

Reply:
I've been primarily referring to NFT as the contract as a whole and tokens as the individual pieces of ownership within the contract.